sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -609,19 +612,19 @@ rowFormat
     ;
 
 multipartIdentifier
-    : parts+=identifier ('.' parts+=identifier)*
+    : parts+=errorCapturingIdentifier ('.' parts+=errorCapturingIdentifier)*
     ;
 
 tableIdentifier
-    : (db=identifier '.')? table=identifier
+    : (db=errorCapturingIdentifier '.')? table=errorCapturingIdentifier
     ;
 
 functionIdentifier
-    : (db=identifier '.')? function=identifier
+    : (db=errorCapturingIdentifier '.')? function=errorCapturingIdentifier
     ;
 
 namedExpression
-    : expression (AS? (identifier | identifierList))?
+    : expression (AS? (name=errorCapturingIdentifier | identifierList))?
```

Comment:
is it possible to break `1 AS a-b`? We can interpret it as `(1 AS a) - b`, which is valid if b is a valid column.

Reply:
I just tried on Postgres, and it's a syntax error.

Reply:
ah nvm, `AS` means the end of an expression in the select list, so only `1 AS a, ...` is allowed

Reply:
In fact, in ANSI-SQL, select statements in a query spec are only allowed to be the format of `expr1 AS alias2, expr2 AS alias2`.

Reply:
FYI https://ronsavage.github.io/SQL/sql-2003-2.bnf.html#select%20list